server/src/main/java/org/opensearch/LegacyESVersion.java

```diff
@@ -135,6 +141,60 @@
     public static final LegacyESVersion V_7_10_2 = new LegacyESVersion(7100299, org.apache.lucene.util.Version.LUCENE_8_7_0);
     public static final LegacyESVersion V_7_10_3 = new LegacyESVersion(7100399, org.apache.lucene.util.Version.LUCENE_8_7_0);
 
+    // todo move back to Version.java if retiring legacy version support
+    protected static final ImmutableOpenIntMap<Version> idToVersion;
+    protected static final ImmutableOpenMap<String, Version> stringToVersion;
+    static {
```

Comment:
What's the reason behind moving the code to here? 
I think we can refactor this to a separate class e.g. `VersionCache` or `CachedVersions`?

Reply:
classloader race condition. The static block requires fields of `LegacyESVersion` to be non-null but `Version` is loaded before `LegacyESVersion` since `LegacyESVersion` inherits from `Version`.

Reply:
PRed https://github.com/opensearch-project/OpenSearch/pull/608 that may have made this easier to catch.